src/operations/blas3/gemm_partial_local.hpp

```diff
@@ -0,0 +1,465 @@
+/***************************************************************************
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename gemm_partial_local.hpp
+ *
+ **************************************************************************/
+
+#ifndef SYCL_BLAS_BLAS3_PARTIAL_GEMM_HPP
+#define SYCL_BLAS_BLAS3_PARTIAL_GEMM_HPP
+
+#include "gemm_common.hpp"
+
+namespace blas {
+
+template <typename input_t, typename output_t, bool DoubleBuffer, bool NbcA,
+          bool NbcB, int ClSize, typename tile_type, bool TransA, bool TransB,
+          typename element_t>
+class GemmPartial<input_t, output_t, DoubleBuffer, NbcA, NbcB, ClSize,
+                  tile_type, TransA, TransB, element_t,
+                  static_cast<int>(gemm_memory_t::local)> {
+ public:
+  using index_t = typename std::make_signed<typename input_t::index_t>::type;
+  using value_t = element_t;
+
+  input_t a_;
+  input_t b_;
+  output_t cube_;
+
+  element_t alpha;
+
+  /* Matrix dimensions */
+  const index_t m_;
```

Comment:
They are read from the views. Is there any reason to pass them as an argument instead of using the matrix views? It seems better if the gemm partial isn't given the wrapper

Reply:
See related answer below